DevTools/HackStudio/LanguageClients/Cpp/ServerConnection.h

```diff
@@ -26,32 +26,37 @@
 
 #pragma once
 
+#include "../../LanguageClient.h"
 #include <AK/LexicalPath.h>
-#include <DevTools/HackStudio/LanguageServers/Cpp/CppLanguageClientEndpoint.h>
-#include <DevTools/HackStudio/LanguageServers/Cpp/CppLanguageServerEndpoint.h>
+#include <DevTools/HackStudio/LanguageServers/LanguageClientEndpoint.h>
+#include <DevTools/HackStudio/LanguageServers/LanguageServerEndpoint.h>
 #include <LibIPC/ServerConnection.h>
 
 namespace LanguageClients {
 namespace Cpp {
 
-class ServerConnection : public IPC::ServerConnection<CppLanguageClientEndpoint, CppLanguageServerEndpoint>
-    , public CppLanguageClientEndpoint {
+class ServerConnection : public HackStudio::ServerConnection
```

Comment:
Now that we have all LSP servers and clients exposing the same IPC endpoints, I wonder if we still need a separate `ServerConnection.h` for every language?
We can have HackStudio::ServerConnection also inherit from `LanguageClientEndpoint` and implement `handshake()` & `handle(AutoCompleteSuggestions)` (And probably also get a path to the local socket of the server in its constructor).

We probably don't want to do something similar with langauge servers though, because we still want to allow flexability in the server-side e.g a language server knows how to update its internal representation of the codebase when `FileEditInsertText` is invoked.

Reply:
Done.